### PR TITLE
Add implicits for EitherT and OptionT transformers

### PR DESCRIPTION
**Pull request by tdbgamer**
@ChristopherDavenport Let me know if there are any issues with this. Library is great so far, but I end up making these instances in my projects so I figured it'd make more sense to contribute it here if you're okay with it.

### CODE REVIEW

**Review by rossabaker — APPROVED** [review]
I'm new here, but these look reasonable to me.

**Comment by tdbgamer** [comment]
Awesome, so is there anything else I need to do before merge?

**Comment by rossabaker** [comment]
I was being cautious about merging to see if another maintainer wanted to weigh in, but I don't think these should be controversial.

I'm going to be working on releases for the move to Typelevel, with full Dotty support, and a CE3 upgrade on a second branch.  I don't have rights to deploy a patch version under Chris' org before The Big move.

**Pull request merged**
